ddcz/forms/authentication.py

```diff
@@ -38,3 +40,197 @@ def get_users(self, email):
+        ),
+    )
+
+    kamaradi_na_webu = forms.CharField(
+        required=False,
+        label="",
+        widget=forms.Textarea(
+            attrs={
+                "placeholder": "Znáš ve městě někoho mimo Virga?",
+                "id": "web_friends",
+                "class": "form-control",
+            }
+        ),
+    )
+
+    odkud_znas_web = forms.CharField(
+        required=False,
+        label="",
+        widget=forms.Textarea(
+            attrs={
+                "placeholder": "Odkud ses dozvěděl o existenci tohoto města (webu)?",
+                "id": "source",
+            }
+        ),
+    )
+
+    gdpr = forms.ChoiceField(
+        label="",
+        widget=forms.Select(attrs={"id": "gdpr"}),
+        choices=GDPR_CHOICES.items(),
+    )
+
+    class Meta:
+        model = UzivateleCekajici
+        fields = [
+            "nick_uzivatele",
+            "email",
+            "jmeno",
+            "prijmeni",
+            "pohlavi",
+            "osloveni",
+            "datum",
+            "patron",
+            "primluvy",
+            "popis_text",
+        ]
+
+    # Setting the data
+    @classmethod
+    def set_for_save(cls, data):
+        data["datum"] = int(datetime.now().timestamp())
+
+        data["primluvy"] = 0
+        data["patron"] = 0
+        data["popis_text"] = cls.set_description(data)
+        return data
+
+    @staticmethod
+    def set_description(data):
+        popis = "1. " + data["duvod_registrace"] + "\n"
+        popis += "2. " + data["odkud_znas_web"] + "\n"
+        popis += "3. " + data["kamaradi_na_webu"] + "\n"
+        popis += "4. \n"
+        popis += "5. \n"
+        return popis
+
+    # Validations for the form
+    def clean_nick_uzivatele(self, *args, **kwargs):
+        nick_uzivatele = self.cleaned_data.get("nick_uzivatele")
+        bad_characters = []
+        for char in self.FORBIDDEN_NICK_CHARACTERS:
+            if char in nick_uzivatele:
+                bad_characters.append(char)
+        if bad_characters.__len__() > 0:
+            raise forms.ValidationError(
+                "Ve vašem nicku jsou některé ze zakázaných znaků: ("
+                + ",".join(bad_characters)
+                + ")."
+            )
+        return nick_uzivatele
+
+    def clean_vek(self, *args, **kwargs):
+        vek = self.cleaned_data.get("vek")
+        if vek < self.MIN_AGE:
+            raise forms.ValidationError(
+                "Bohužel ti není " + self.MIN_AGE + " let. Zkus to, až budeš starší."
+            )
+        return vek
+
+    def clean_pohlavi(self, *args, **kwargs):
+        pohlavi = self.cleaned_data.get("pohlavi")
+        if pohlavi not in ["0", "1"]:
+            raise forms.ValidationError(
+                "Bohužel toto pohlaví nám zůstává utajeno. Neplete si pohlaví s genderem?"
+            )
+        return pohlavi
+
+    def clean_gdpr(self, *args, **kwargs):
+        gdpr = self.cleaned_data.get("gdpr")
+        if gdpr is not "1":
```

Comment:
Just of note, `!=` would be better here. The fact that `is not` works is an accidental byproduct of how CPython implements strings :)

Reply:
And I thought I was being more "python" here. :D